src/runtime/scheduler/scheduler.rs

```diff
@@ -55,40 +55,27 @@ impl Scheduler {
         self.groups.insert(TaskGroup::default()).into()
     }
 
-    fn get_mut_group(&mut self, id: SchedulerId) -> Option<&mut TaskGroup> {
+    fn get_group_mut(&mut self, id: SchedulerId) -> Option<&mut TaskGroup> {
         self.groups.get_mut(id.into())
     }
 
-    /// The parent id can either be the id of the group or another task in the same group.
-    pub fn insert_and_poll_task<T: Task>(&mut self, group_id: SchedulerId, task: T) -> Option<InsertResult> {
-        let group: &mut TaskGroup = self.get_mut_group(group_id)?;
-        group.insert_and_poll_task(Box::new(task))
+    pub fn insert<T: Task>(&mut self, group_id: SchedulerId, task: T) -> Option<InsertResult> {
```

Comment:
I think we might want to name this insert_and_run task, just to make it really clear that we are going to trigger the task right away.

Reply:
Changed all the related fns to be `schedule*` so that it flows naturally with the functionality.